pkg/controller/releasemanager/scaling/geometric.go

```diff
@@ -0,0 +1,42 @@
+package scaling
+
+import "time"
+
+func GeometricScale(st ScalableTarget, max uint32, t time.Time) uint32 {
+	current := st.CurrentPercent()
+	release := st.ReleaseInfo()
+
+	if max <= 0 {
+		return 0
+	}
+
+	if release.Max < max {
+		max = release.Max
+	}
+
+	// Scaling down
+	if current > max {
+		return max
+	}
+
+	// Wait longer to scale
+	deadline := st.LastUpdated().Add(release.GeometricScaling.Delay.Duration)
+	if deadline.After(t) {
+		return current
+	}
+
+	desired := current * release.GeometricScaling.Factor
+	if current < release.GeometricScaling.Start {
+		desired = release.GeometricScaling.Start
+	}
+
+	if desired > max {
+		desired = max
+	}
+
+	if desired == current || st.IsReconciled(desired) {
+		return desired
+	}
+
+	return current
```

Comment:
```suggestion
	// Return the new desired scale if the scalable target is ready for it.
	if st.IsReconciled(desired) {
		return desired
	}

	// Otherwise, hold at the current scale.
	return current
```

I had to look up what `IsReconciled` actually does to understand what this does and what it’s correct (it checks pretty much the opposite of what I guessed).

Also there’s no need to check `desired == current` since in that case, the two branches do the same thing.

(same goes for the identical parts of`linear.go`, too)

Reply:
Removed noxious code and renamed function to be more clear.